The GH Windows failure test logs are not available.  The Pipelines  logs are and same on win32 and win64
======================================================================
FAIL: test_map (tkinter.test.test_ttk.test_style.StyleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\a\1\s\lib\tkinter\test\test_ttk\test_style.py", line 45, in test_map
    self.assertEqual(m['background'], [('active', 'white')])
AssertionError: Lists differ: ['active', 'white'] != [('active', 'white')]
======================================================================
FAIL: test_map_custom_copy (tkinter.test.test_ttk.test_style.StyleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\a\1\s\lib\tkinter\test\test_ttk\test_style.py", line 184, in test_map_custom_copy
    self.assertEqual(style.map(newname, key), value)
AssertionError: Lists differ: [('!disabled pressed', 'sunken')] != [('!disabled', 'pressed', 'sunken')]

--- a/Lib/tkinter/test/test_ttk/test_style.py
+++ b/Lib/tkinter/test/test_ttk/test_style.py
@@ -37,45 +37,33 @@ class StyleTest(AbstractTkTest, unittest.TestCase):
 
         # Single state
         for states in ['active'], [('active',)]:
-            style.map('TButton', background=[('active', 'white')])
-            self.assertEqual(style.map('TButton', 'background'), [('active', 'white')])
-            m = style.map('TButton')
-            self.assertIsInstance(m, dict)
-            if self.wantobjects:
-                self.assertEqual(m['background'], [('active', 'white')])
-            else:
-                self.assertEqual(m['background'], 'active white')
+            with self.subTest(states=states):
+                style.map('TButton', background=[(*states, 'white')])
+                expected = [('active', 'white')]
+                self.assertEqual(style.map('TButton', 'background'), expected)
+                m = style.map('TButton')
+                self.assertIsInstance(m, dict)
+                self.assertEqual(m['background'], expected)
 
         # Multiple states
-        for states in ['pressed', '!disabled'], [('pressed', '!disabled')], ['pressed !disabled']:
-            style.map('TButton', background=[(*states, 'black')])
-            if self.wantobjects:
-                self.assertEqual(style.map('TButton', 'background'),
-                                [('pressed', '!disabled', 'black')])
-            else:
-                self.assertEqual(style.map('TButton', 'background'),
-                                [('pressed !disabled', 'black')])
-            m = style.map('TButton')
-            self.assertIsInstance(m, dict)
-            if self.wantobjects:
-                self.assertEqual(m['background'],
-                                [('pressed', '!disabled', 'black')])
-            else:
-                self.assertEqual(m['background'], '{pressed !disabled} black')
+        for states in ['pressed', '!disabled'], ['pressed !disabled'], [('pressed', '!disabled')]:
+            with self.subTest(states=states):
+                style.map('TButton', background=[(*states, 'black')])
+                expected = [('pressed', '!disabled', 'black')]
+                self.assertEqual(style.map('TButton', 'background'), expected)
+                m = style.map('TButton')
+                self.assertIsInstance(m, dict)
+                self.assertEqual(m['background'], expected)
 
         # Default state
-        for states in [], ['']:
-            style.map('TButton', background=[(*states, 'grey')])
-            if self.wantobjects:
-                self.assertEqual(style.map('TButton', 'background'), [('grey',)])
-            else:
-                self.assertEqual(style.map('TButton', 'background'), [('', 'grey')])
-            m = style.map('TButton')
-            self.assertIsInstance(m, dict)
-            if self.wantobjects:
-                self.assertEqual(m['background'], [('grey',)])
-            else:
-                self.assertEqual(m['background'], '{} grey')
+        for states in [], [''], [()]:
+            with self.subTest(states=states):
+                style.map('TButton', background=[(*states, 'grey')])
+                expected = [('grey',)]
+                self.assertEqual(style.map('TButton', 'background'), expected)
+                m = style.map('TButton')
+                self.assertIsInstance(m, dict)
+                self.assertEqual(m['background'], expected)
 
 
     def test_lookup(self):
@@ -135,53 +123,47 @@ class StyleTest(AbstractTkTest, unittest.TestCase):
 
 
     def test_configure_custom_copy(self):
-        if not self.wantobjects:
-            self.skipTest("requires wantobjects=1")
-
         style = self.style
 
         curr_theme = self.style.theme_use()
         self.addCleanup(self.style.theme_use, curr_theme)
         for theme in self.style.theme_names():
             self.style.theme_use(theme)
-
             for name in CLASS_NAMES:
                 default = style.configure(name)
                 if not default:
                     continue
-                if support.verbose >= 2:
-                    print('configure', theme, name, default)
-                newname = f'C.{name}'
-                self.assertEqual(style.configure(newname), None)
-                style.configure(newname, **default)
-                self.assertEqual(style.configure(newname), default)
-                for key, value in default.items():
-                    self.assertEqual(style.configure(newname, key), value)
+                with self.subTest(theme=theme, name=name):
+                    if support.verbose >= 2:
+                        print('configure', theme, name, default)
+                    newname = f'C.{name}'
+                    self.assertEqual(style.configure(newname), None)
+                    style.configure(newname, **default)
+                    self.assertEqual(style.configure(newname), default)
+                    for key, value in default.items():
+                        self.assertEqual(style.configure(newname, key), value)
 
 
     def test_map_custom_copy(self):
-        if not self.wantobjects:
-            self.skipTest("requires wantobjects=1")
-
         style = self.style
 
         curr_theme = self.style.theme_use()
         self.addCleanup(self.style.theme_use, curr_theme)
         for theme in self.style.theme_names():
             self.style.theme_use(theme)
-
             for name in CLASS_NAMES:
                 default = style.map(name)
                 if not default:
                     continue
-                if support.verbose >= 2:
-                    print('map', theme, name, default)
-                newname = f'C.{name}'
-                self.assertEqual(style.map(newname), {})
-                style.map(newname, **default)
-                self.assertEqual(style.map(newname), default)
-                for key, value in default.items():
-                    self.assertEqual(style.map(newname, key), value)
+                with self.subTest(theme=theme, name=name):
+                    if support.verbose >= 2:
+                        print('map', theme, name, default)
+                    newname = f'C.{name}'
+                    self.assertEqual(style.map(newname), {})
+                    style.map(newname, **default)
+                    self.assertEqual(style.map(newname), default)
+                    for key, value in default.items():
+                        self.assertEqual(style.map(newname, key), value)
 
 
 tests_gui = (StyleTest, )

--- a/Lib/tkinter/ttk.py
+++ b/Lib/tkinter/ttk.py
@@ -248,6 +248,8 @@ def _list_from_statespec(stuple):
     for state, val in zip(it, it):
         if hasattr(state, 'typename'):  # this is a Tcl object
             state = str(state).split()
+        elif isinstance(state, str):
+            state = state.split()
         elif not isinstance(state, (tuple, list)):
             state = (state,)
         if hasattr(val, 'typename'):
@@ -394,13 +396,12 @@ class Style(object):
         or something else of your preference. A statespec is compound of
         one or more states and then a value."""
         if query_opt is not None:
-            return _list_from_statespec(self.tk.splitlist(
-                self.tk.call(self._name, "map", style, '-%s' % query_opt)))
+            result = self.tk.call(self._name, "map", style, '-%s' % query_opt)
+            return _list_from_statespec(self.tk.splitlist(result))
 
-        return _splitdict(
-            self.tk,
-            self.tk.call(self._name, "map", style, *_format_mapdict(kw)),
-            conv=_list_from_statespec)
+        result = self.tk.call(self._name, "map", style, *_format_mapdict(kw))
+        return {k: _list_from_statespec(self.tk.splitlist(v))
+                for k, v in _splitdict(self.tk, result).items()}
 
 
     def lookup(self, style, option, state=None, default=None):

--- a/Misc/NEWS.d/next/Library/2020-11-15-17-02-00.bpo-42328.bqpPlR.rst
+++ b/Misc/NEWS.d/next/Library/2020-11-15-17-02-00.bpo-42328.bqpPlR.rst
@@ -1,2 +1,4 @@
-Fixed :meth:`tkinter.ttk.Style.map`. It accepts now the default state as
-returned by ``Style.map()``.
+Fixed :meth:`tkinter.ttk.Style.map`. The function accepts now the
+representation of the default state as empty sequence (as returned by
+``Style.map()``). The structure of the result is now the same on all platform
+and does not depend on the value of ``wantobjects``.

